src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1138,6 +1138,34 @@ public FileDialog GetSaveDialog(WorkspaceModel workspace)
         }
 
         /// <summary>
+        /// Attempts to open a file using the passed open command, but wraps the call
+        /// with a check to make sure no unsaved changes to the HomeWorkspace are lost.
+        /// </summary>
+        /// <param name="openCommand"> <see cref="DynamoModel.OpenFileCommand"/> </param>
+        private void OpenIfSaved(object openCommand)
+        {
```

Comment:
I don't want to change the behavior of the openFileCommand, I tested this first and it alters behavior in a bunch of places... throws multiple dialogs, etc etc.